File: cpp/timeConversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the timeConversion function below.
 */
string timeConversion(string s) {
    /*
     * Write your code here.
     */
    
    if(s.substr(8,10) == "PM"){
        int hours = stoi(s.substr(0,2));
        if(hours != 12){
            hours += 12;
        }
        s[0] = to_string(hours)[0];
        s[1] = to_string(hours)[1];
        return s.substr(0,8);
    }else if(s.substr(0,2) == "12"){
        return "00" + s.substr(2,6);
    }else{
        return s.substr(0,8);
    }
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Python Solutions/timeConversion.py
#!/bin/python3

import os
import sys

#
# Complete the timeConversion function below.
#
def timeConversion(s):
    #
    # Write your code here.
    #
    if s[8:] == "PM":
        hours = int(s[:2])
        if hours != 12:
            hours += 12
        s = str(hours) + s[2:8]
        return s
    elif s[:2] == "12":
        s = "00" + s[2:8]
        return s
    else:
        return s[:8]

if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    f.write(result + '\n')

    f.close()
